### PR TITLE
EPMRPP-33463 React. Dashboard. Widget. Passing rate per launch/ summary

### PR DESCRIPTION
**Pull request by tamas-levente-szucs-epam**
bar-pie

### CODE REVIEW

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+      });
+      return;
+    }
+
+    this.node.addEventListener('mousemove', this.getCoords);
+
+    if (!this.props.isPreview) {
+      this.resizeHelper(this.node);
+    }
+    // this.resizeChart();
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx]: ?

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+      axis,
+      bar,
+    };
+  };
+
+  getConfig = () => {
+    const { widget, isPreview, container } = this.props;
+    const data = widget.content.result;
+    const colors = {
+      Passed: COLORS[`COLOR_PASSED_PER_LAUNCH`],
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx]: `COLORS.COLOR_PASSED_PER_LAUNCH` or `COLORS[COLOR_PASSED_PER_LAUNCH]`?

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      [
+        'Failed',
+        parseInt(data.content.result.total, 10) - parseInt(data.content.result.passed, 10),
+      ],
+    ];
+    const columnData = { Passed: columns[0][1], Failed: columns[1][1] };
+    const chartData = {
+      columns,
+      groups: [itemNames],
+      type: data.content_parameters.widgetOptions.viewMode === 'barMode' ? 'bar' : 'pie',
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx]: `contentParameters`
WS is switching to camel case everywhere

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+      this.state.isConfigReady && (
+        <C3Chart
+          config={this.config}
+          onChartCreated={this.onChartCreated}
+          className={this.props.widget.content_parameters.widgetOptions.viewMode[0]}
+        >
+          <Legend
+            isPreview={this.props.isPreview}
+            items={this.itemData.itemNames}
+            onClick={() => {}}
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx]: This prop defaults to `() => {}`, so you may omit it

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
+:global {
+  .c3-chart-text text,
+  .c3-chart-arc text {
+    font-family: OpenSansSemibold, sans-serif;
+    font-size: 13px;
+    fill: #fff !important;
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.scss]: variable

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx [hunk 6]
```diff
@@ -30,6 +31,9 @@ const charts = {
   [widgetTypes.FAILED_CASES_TREND]: FailedCasesTrendChart,
   [widgetTypes.NON_PASSED_TEST_CASES_TREND]: NonPassedTestCasesTrendChart,
   [widgetTypes.TEST_CASES_GROWTH_TREND]: TestCasesGrowthTrendChart,
+  launches_comparison_chart: LaunchesComparisonChart,
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx]: Please use `widgetTypes`

File: app/src/common/constants/colors.js [hunk 7]
```diff
@@ -10,3 +10,5 @@ export const COLOR_TO_INVESTIGATE = '#ffb743';
 export const COLOR_CHART_DURATION = '#557fd3';
 export const COLOR_ORANGE_RED = '#ff3222';
 export const COLOR_DARK_PASTEL_GREEN = '#5cb755';
+export const COLOR_PASSED_PER_LAUNCH = '#8db677';
```
[the remark this answers is not in the document]
Comment (tamas-levente-szucs-epam) [reply app/src/common/constants/colors.js]: this colors was in the old version. 
export const COLOR_PASSED_PER_LAUNCH = '#8db677';
export const COLOR_FAILED_PER_LAUNCH = '#e86c42';
Who is the designer? I disguss it with them.

[the remark this answers is not in the document]
Comment (DiscoElevator) [reply app/src/common/constants/colors.js]: https://github.com/reportportal/service-ui/blob/v5/app/docs/2-responsiblePersons.md

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx (hunk 3 above)
Comment (tamas-levente-szucs-epam) [reply app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx]: This content_parameters comes from the demodata. 
U suggesting to converting to it to camel case? Or the backend give us in the future camel case json?

Comment (DiscoElevator) [reply app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.jsx]: In the near future all responses will be in camel case.

File: app/src/common/constants/colors.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (tamas-levente-szucs-epam) [reply app/src/common/constants/colors.js]: done. i speak with Sasha. She suggest me 2 other colors to use here.

File: app/src/components/widgets/charts/common/legend/legend.scss [hunk 8]
```diff
@@ -45,3 +44,30 @@
...
+  overflow: hidden;
+  top: 0;
+  width: calc(100% - 40px);
+  height: 85px;
+  .info-data {
+    width: 70%;
+    clear: both;
+    text-align: left;
+    margin-bottom: 10px;
+    font-family: OpenSansSemibold, sans-serif;
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/common/legend/legend.scss]: `$FONT-SEMIBOLD`

File: app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.scss [hunk 9]
```diff
@@ -0,0 +1,11 @@
+:global {
+  .c3-chart-text text,
+  .c3-chart-arc text {
+    font-family: OpenSansSemibold, sans-serif;
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/passingRatePerLaunch.scss]: `$FONT-SEMIBOLD`

File: app/src/components/widgets/charts/passingRatePerLaunch/demoData.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
+export const passingRateBarData = {
+  share: false,
+  id: 2,
+  name: 'test widget1',
+  applied_filters: [],
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/passingRatePerLaunch/demoData.js]: Please use camel case

File: app/src/components/widgets/charts/common/legend/legend.jsx [hunk 11]
```diff
@@ -52,23 +52,39 @@ const messages = defineMessages({
     id: 'Widgets.ofTestCases',
     defaultMessage: 'of test cases',
   },
+  Passed: {
+    id: 'Passed',
+    defaultMessage: 'Passed',
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/common/legend/legend.jsx]: Please add translations

https://github.com/reportportal/service-ui/blob/v5/app/docs/10-localization.md

**Pull request merged**